### DISCUSSION TITLE
Welcome to HomeAutomation.CoDeSys3 Discussions!

### DISCUSSION THREAD

**Discussion by MichielVanwelsenaere** [MichielVanwelsenaere/HomeAutomation.CoDeSys3#79] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by beal1** [comment]
Hi Michiel, 

I'm a newbie in GH an HA, 
So what are hardware need, 
I've a 750-8212 and a lot of DI DO 
i'm using codesys V2.3
what about HA ? does I need a (fanless mini pc for example ?) or the PFC can run it ?
I supposed that HA is capable to read and write from "smart life appliance"or "lidl hub" 

regards 
Al1

**Reply by MichielVanwelsenaere** [reply]
Hi,

you'll need to switch to codesys3.5 to be able to run this project.
In you can try running HA on the PFC but I'm not sure I would recommend it.

In terms of recommended hardware for HA and integration with "smart life appliance"or "lidl hub" I'm afraid you'll need to do some digging on your own ;)

**Reply by beal1** [reply]
thanks, 

i'll continue the physical wiring and start codesys 2.3 programming 
once this layer is working, go further investigation with MQTT and HA 

I'll be later back :)

**Comment by ElAmarok** [comment]
Hello Mr. Vanwelsenaere,

I am using at the moment Codesys3.5 with IoBroker connected with ModbusTCP.
Now i want to switch to Home Assistant. I found your Librarys in the HA forum wanted to try MQTT instead ModbusTCP now.
Let me tell you first, i am not very experienced in programming, I am just a standard maintanance electrican in a big company.

Unfortunately, I haven't yet figured out how it would be possible to implement your function blocks in an existing project. I don't understand what dependencies are necessary or how best to start. I followed your instructions exactly by creating a new project as shown in the instructions, but I can't figure out how I can incorporate all of this into my project and use your great function blocks (FB_INPUT_BINARYSENSOR_MQTT, FB_OUTPUT_BINARY_MQTT,...) . Do you perhaps have a tip for me on how I could proceed?
Greetings from Austria,
Patrick

**Reply by primsam** [reply]
Hello Patrick,

Hello from the south side of the alps.
If you want i can give you some Support how to use that great project.

Samuel

**Reply by MichielVanwelsenaere** [reply]
Hi Patrick,

The projects function blocks aren't really build to be drag and drop from project A to project B. The whole project should be considered as one unit. The reason the function blocks aren't drag and drop is that they require mqtt publish and subscribe logic which is not encapsulated in the function blocks itself.

No need to be an experienced programmer, several people with limited experience have gotten this to work for their use cases. and you've already had someone offer you help ;). Thanks @primsam!

I think it's worth considering starting over your implementation using the sample project provided in this project. The main question to that aspect is; are you able to test the new software on your current setup without causing to much frustration in your household..? :).

Good luck!

**Comment by ElAmarok** [comment]
Thanks so much for your reply's!

The best thing in my situation is: I have a complete Testsystem in my Homeoffice. Just for learning.
![P_20240214_190107](https://github.com/MichielVanwelsenaere/HomeAutomation.CoDeSys3/assets/33393715/02c5fa48-a752-48d9-b156-2e191d8a6c73)

But unfortunately that doesn't make me a better programmer.

@primsam 
Thanks for your time.
If I have already followed the instructions from Mr. Vanwelsenaere. What would be the next step? or should I run a second application with the Mqtt part from Mr. Vanwelsenaere on the controller? What would be the correct course of action?
Greeting Patrick

**Reply by primsam** [reply]
Hello Patrick,

I can offer you some assistance over remote connection and or a call. That would be the easiest way I think.
So I can first figure out how far you already are and give you the right hints 🙂.
Does anybody know to send private messages here ?
Because I don't want to put my email address here publicly.😅

Samuel

**Reply by MichielVanwelsenaere** [reply]
you can't over here. I would suggest using Discord and passing the username over here.

**Comment by ElAmarok** [comment]
Remote connection sounds great. Would Teamviewer be suitable for something like this? I'm also finally going to take a look at this Discord. I've never used this before. 
BUT: 
1. I want to try for a while to see how far I can get myself. 
2. My entire family, including me, is pretty sick right now. In times when I'm not suffering, I keep taking a look at the program, but it's not very effective. 

It would help me a lot if I can get back in touch via Git Hub when everything is back to normal and I have already made progress with the program. 
Would that be okay?
